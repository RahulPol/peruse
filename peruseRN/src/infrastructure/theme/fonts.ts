import { FontWeight } from '../typings/fonts';

export const fontConfig = {
  ios: {
    regular: {
      fontFamily: 'Roboto_400Regular',
      fontWeight: '400' as FontWeight,
    },
    medium: {
      fontFamily: 'Roboto_500Medium',
      fontWeight: '500' as FontWeight,
    },
    light: {
      fontFamily: 'Roboto_300Light',
      fontWeight: '300' as FontWeight,
    },
    thin: {
      fontFamily: 'Roboto_100Thin',
      fontWeight: '100' as FontWeight,
    },
    bold: {
      fontFamily: 'Roboto_700Bold',
      fontWeight: '700' as FontWeight,
    },
  },
  android: {
    regular: {
      fontFamily: 'Roboto_400Regular',
      fontWeight: '400' as FontWeight,
    },
    medium: {
      fontFamily: 'Roboto_500Medium',
      fontWeight: '500' as FontWeight,
    },
    light: {
      fontFamily: 'Roboto_300Light',
      fontWeight: '300' as FontWeight,
    },
    thin: {
      fontFamily: 'Roboto_100Thin',
      fontWeight: '100' as FontWeight,
    },
    bold: {
      fontFamily: 'Roboto_700Bold',
      fontWeight: '700' as FontWeight,
    },
  },
};
