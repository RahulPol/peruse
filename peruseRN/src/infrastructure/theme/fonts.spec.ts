import { fontSizes, fontWeights, fonts } from './fonts';

describe('fontConfig', () => {
  it('should match the snapshot', () => {
    expect(fonts).toMatchSnapshot();
    expect(fontWeights).toMatchSnapshot();
    expect(fontSizes).toMatchSnapshot();
  });
});
