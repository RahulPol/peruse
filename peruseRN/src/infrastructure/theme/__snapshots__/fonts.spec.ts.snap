// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fontConfig should match the snapshot 1`] = `
Object {
  "body": "Oswald_400Regular",
  "heading": "Lato_400Regular",
  "monospace": "Oswald_400Regular",
}
`;

exports[`fontConfig should match the snapshot 2`] = `
Object {
  "bold": 700,
  "medium": 500,
  "regular": 400,
}
`;

exports[`fontConfig should match the snapshot 3`] = `
Object {
  "appTitle": "32px",
  "body": "16px",
  "button": "14px",
  "caption": "12px",
  "h1": "112px",
  "h2": "56px",
  "h3": "45px",
  "h4": "34px",
  "h5": "24px",
  "title": "20px",
}
`;
